RecoParticleFlow/PFProducer/plugins/PFProducer.cc

```diff
@@ -333,3 +309,233 @@ void PFProducer::produce(Event& iEvent, const EventSetup& iSetup) {
+  desc.add<int>(
+      "goodPixelTrackDeadHcal_maxLost3Hit",
+      0);  // max missing outer hits for a track with 3 valid pixel layers (can set to -1 to reject all these tracks)
+  desc.add<int>("goodPixelTrackDeadHcal_maxLost4Hit",
+                1);  // max missing outer hits for a track with >= 4 valid pixel layers
+  desc.add<double>("goodPixelTrackDeadHcal_dxy", 0.02);  // [cm] abs(trackRef->dxy(primaryVertex_.position())) < X
+  desc.add<double>("goodPixelTrackDeadHcal_dz", 0.05);   // [cm] abs(trackRef->dz(primaryVertex_.position())) < X
+
+  // number of sigmas for neutral energy detection
+  desc.add<double>("pf_nsigma_ECAL", 0.0);
+  desc.add<double>("pf_nsigma_HCAL", 1.0);
+
+  // ECAL/HCAL PF cluster calibration : take it from global tag ?
+  desc.add<bool>("useCalibrationsFromDB", true);
+  desc.add<std::string>("calibrationsLabel", "");
+
+  // Post HF cleaning
+  desc.add<bool>("postHFCleaning", false);
+  {
+    edm::ParameterSetDescription psd_PFHFCleaning;
+    // Clean only objects with pt larger than this value
+    psd_PFHFCleaning.add<double>("minHFCleaningPt", 5.0);
+    // Clean only if the initial MET/sqrt(sumet) is larger than this value
+    psd_PFHFCleaning.add<double>("maxSignificance", 2.5);
+    // Clean only if the final MET/sqrt(sumet) is smaller than this value
+    psd_PFHFCleaning.add<double>("minSignificance", 2.5);
+    // Clean only if the significance reduction is larger than this value
+    psd_PFHFCleaning.add<double>("minSignificanceReduction", 1.4);
+    // Clean only if the MET and the to-be-cleaned object satisfy this DeltaPhi * Pt cut
+    // (the MET angular resoution is in 1/MET)
+    psd_PFHFCleaning.add<double>("maxDeltaPhiPt", 7.0);
+    // Clean only if the MET relative reduction from the to-be-cleaned object
+    // is larger than this value
+    psd_PFHFCleaning.add<double>("minDeltaMet", 0.4);  //
+    desc.add<edm::ParameterSetDescription>("PFHFCleaningParameters", psd_PFHFCleaning);
+  }
+
+  // Check HF cleaning
+  desc.add<std::vector<edm::InputTag>>("cleanedHF",
+                                       {
+                                           edm::InputTag("particleFlowRecHitHF", "Cleaned"),
+                                           edm::InputTag("particleFlowClusterHF", "Cleaned"),
+                                       });
+
+  // calibration parameters for HF:
+  desc.add<bool>("calibHF_use", false);
+  desc.add<std::vector<double>>("calibHF_eta_step", {0.0, 2.9, 3.0, 3.2, 4.2, 4.4, 4.6, 4.8, 5.2, 5.4});
+  desc.add<std::vector<double>>("calibHF_a_EMonly", {1., 1., 1., 1., 1., 1., 1., 1., 1., 1.});
+  desc.add<std::vector<double>>("calibHF_a_EMHAD", {1., 1., 1., 1., 1., 1., 1., 1., 1., 1.});
+  desc.add<std::vector<double>>("calibHF_b_HADonly", {1., 1., 1., 1., 1., 1., 1., 1., 1., 1.});
+  desc.add<std::vector<double>>("calibHF_b_EMHAD", {1., 1., 1., 1., 1., 1., 1., 1., 1., 1.});
```

Comment:
these are all different from what was in particleFlow_cfi.py before.
Is this intentional?
is this hardcoded calib machinery still needed?

Reply:
These are not, but I didn't want to completely remove them. So, parameters are just set like these.